crates/ruff/src/checkers/ast/mod.rs

```diff
@@ -2201,7 +2201,12 @@ where
                     }
```

Comment:
Nit: this part above can be rewritten with some optional chaining:

```rust
let check = self
    .semantic
    .expr_grandparent()
    .and_then(Expr::as_subscript_expr)
    .map_or(true, |parent| {
        !self.semantic.match_typing_expr(&parent.value, "Union")
    });
```

This is nice because you can avoid the `mut`, and all the logic is contained in one fluid expression.

Reply:
Nit: I would prefer keeping the variable (I find such lengthy if's hard to read) but give it a name with more semantical meaning. Like, what are we testing here? `is_union` maybe?

This has the advantage that if there's a bug I have an easier time fixing it because:

* if `is_union` is `false` when the `expression` is a union -> Okay, the bug must be with the expression initializing `is_union`. The same if `is_union` is `true` when it should not.
* I don't need to spend time understanding this expression if the bug is with the diagnostic or fix generation but not identifying what a union is.

Reply:
`is_union_but_grandparent_is_not` :) I'm not sure what a good name would be yet. I'll think on it.